src/server/lib/okta/oidc.ts

```diff
@@ -0,0 +1,272 @@
+  introspection_endpoint: `${oktaDomain}/oauth2/${oktaCustomOAuthServer}/v1/introspect`,
+  revocation_endpoint: `${oktaDomain}/oauth2/${oktaCustomOAuthServer}/v1/revoke`,
+  end_session_endpoint: `${oktaDomain}/oauth2/${oktaCustomOAuthServer}/v1/logout`,
+  claim_types_supported: ['normal'],
+  claims_parameter_supported: false,
+  grant_types_supported: [
+    'authorization_code',
+    'implicit',
+    'refresh_token',
+    'password',
+  ],
+  request_parameter_supported: true,
+  request_uri_parameter_supported: true,
+  require_request_uri_registration: false,
+  response_modes_supported: [
+    'query',
+    'fragment',
+    'form_post',
+    'okta_post_message',
+  ],
+  token_endpoint_auth_methods_supported: [
+    'client_secret_basic',
+    'client_secret_post',
+    'client_secret_jwt',
+    'private_key_jwt',
+    'none',
+  ],
+  response_types_supported: [
+    'code',
+    'id_token',
+    'code id_token',
+    'code token',
+    'id_token token',
+    'code id_token token',
+  ],
+  subject_types_supported: ['public'],
+  id_token_signing_alg_values_supported: ['RS256'],
+  scopes_supported: [
+    'openid',
+    'profile',
+    'email',
+    'address',
+    'phone',
+    'offline_access',
+  ],
+  claims_supported: [
+    'iss',
+    'ver',
+    'sub',
+    'aud',
+    'iat',
+    'exp',
+    'jti',
+    'auth_time',
+    'amr',
+    'idp',
+    'nonce',
+    'name',
+    'nickname',
+    'preferred_username',
+    'given_name',
+    'middle_name',
+    'family_name',
+    'email',
+    'email_verified',
+    'profile',
+    'zoneinfo',
+    'locale',
+    'address',
+    'phone_number',
+    'picture',
+    'website',
+    'gender',
+    'birthdate',
+    'updated_at',
+    'at_hash',
+    'c_hash',
+  ],
+  code_challenge_methods_supported: ['S256'],
+  introspection_endpoint_auth_methods_supported: [
+    'client_secret_basic',
+    'client_secret_post',
+    'client_secret_jwt',
+    'private_key_jwt',
+    'none',
+  ],
+  revocation_endpoint_auth_methods_supported: [
+    'client_secret_basic',
+    'client_secret_post',
+    'client_secret_jwt',
+    'private_key_jwt',
+    'none',
+  ],
+  request_object_signing_alg_values_supported: [
+    'HS256',
+    'HS384',
+    'HS512',
+    'RS256',
+    'RS384',
+    'RS512',
+    'ES256',
+    'ES384',
+    'ES512',
+  ],
+};
+
+// Encapsulates a discovered or instantiated OpenID Connect Issuer (Issuer),
+// Identity Provider(IdP), Authorization Server(AS) and its metadata.
+const OIDCIssuer = new Issuer(OIDC_METADATA);
+
+// Encapsulates a dynamically registered, discovered or instantiated
+// OpenID Connect Client(Client), Relying Party(RP), and its metadata,
+// its instances hold the methods for getting an authorization URL,
+// consuming callbacks, triggering token endpoint grants, revoking and
+// introspecting tokens.
+// this is set to the "profile" client id and secret
+export const OIDCProfileClient = new OIDCIssuer.Client({
```

Comment:
It might be simpler if we only export the properties we use rather than this whole object? It seems like we don't need to call it very often?